google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb

```diff
@@ -419,8 +440,72 @@ def close
         #   batch_snapshot = batch_client.batch_snapshot
         #
         #   results = batch_snapshot.execute "SELECT * FROM users " \
-        #                         "WHERE active = @active",
-        #                         params: { active: true }
+        #                                    "WHERE active = @active",
+        #                                    params: { active: true }
+        #
+        #   results.rows.each do |row|
+        #     puts "User #{row[:id]} is #{row[:name]}"
+        #   end
+        #
+        # @example Query with a SQL STRUCT query parameter as a Hash:
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #   batch_client = spanner.batch_client "my-instance", "my-database"
+        #   batch_snapshot = batch_client.batch_snapshot
+        #
+        #   user_hash = { id: 1, name: "Charlie", active: false }
+        #
+        #   results = batch_snapshot.execute "SELECT * FROM users WHERE " \
+        #                                    "ID = @user_struct.id " \
+        #                                    "AND name = @user_struct.name " \
+        #                                    "AND active = @user_struct.active",
+        #                                    params: { user_struct: user_hash }
+        #
+        #   results.rows.each do |row|
+        #     puts "User #{row[:id]} is #{row[:name]}"
+        #   end
+        #
+        # @example Specify the SQL STRUCT type using Fields object:
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #   batch_client = spanner.batch_client "my-instance", "my-database"
+        #   batch_snapshot = batch_client.batch_snapshot
+        #
+        #   user_type = batch_client.fields(
+        #     { id: :INT64, name: :STRING, active: :BOOL }
+        #   )
+        #   user_hash = { id: 1, name: nil, active: false }
+        #
+        #   results = batch_snapshot.execute "SELECT * FROM users WHERE " \
+        #                                    "ID = @user_struct.id " \
+        #                                    "AND name = @user_struct.name " \
+        #                                    "AND active = @user_struct.active",
+        #                                    params: { user_struct: user_hash },
+        #                                    types: { user_struct: user_type }
```

Comment:
what happens if they do not pass the type here? How would the library infer the type of "name"?